cfgov/mega_menu/frontend_conversion.py

```diff
@@ -0,0 +1,148 @@
+class FrontendConverter:
+    def __init__(self, menu, request=None):
```

Comment:
This code may seem to be generating arbitrarily named and structured Python dicts for the frontend, and indeed it is, mainly to ensure compatibility with both the legacy and proposed frontend code. In future we can hopefully simplify and rename quite a bit of this (for example avoiding the use of `external_link` when passing menu URLs to the frontend). I wanted to avoid having to alter the existing menu frontend code as part of this PR.